src/common/history.c

```diff
@@ -870,48 +870,106 @@ int dt_history_copy_and_paste_on_selection(int32_t imgid, gboolean merge, GList
   return res;
 }
 
+void dt_set_history_compress_problem(int32_t imgid, gboolean set)
```

Comment:
To follow the naming convention it should be:
dt_history_set_compress_problem()